Failure during build (EventStruct error)

On a fresh install of Arch Linux, and I'm unable to build from the AUR repo:  https://aur.archlinux.org/polybar.git

Here's the console output:

[tyler@mercury ~/builds/polybar]$ makepkg -si
==> Making package: polybar 3.1.0-2 (Sun Mar 11 11:32:14 PDT 2018)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Cloning polybar git repo...
Cloning into bare repository '/home/tyler/builds/polybar/polybar'...
remote: Counting objects: 13709, done.
remote: Compressing objects: 100% (141/141), done.
remote: Total 13709 (delta 149), reused 103 (delta 74), pack-reused 13494
Receiving objects: 100% (13709/13709), 7.19 MiB | 22.30 MiB/s, done.
Resolving deltas: 100% (10252/10252), done.
==> Validating source files with md5sums...
    polybar ... Skipped
==> Extracting sources...
  -> Creating working copy of polybar git repo...
Cloning into 'polybar'...
done.
Switched to a new branch 'makepkg'
==> Starting prepare()...
Submodule 'lib/i3ipcpp' (https://github.com/jaagr/i3ipcpp) registered for path 'lib/i3ipcpp'
Submodule 'lib/xpp' (https://github.com/jaagr/xpp) registered for path 'lib/xpp'
Cloning into '/home/tyler/builds/polybar/src/polybar/lib/i3ipcpp'...
Cloning into '/home/tyler/builds/polybar/src/polybar/lib/xpp'...
Submodule path 'lib/i3ipcpp': checked out 'a6aa7a19786bdf7b96a02900510b3b3c325c8bdf'
Submodule path 'lib/xpp': checked out 'c73db50f3020b7abe294631b25a2d302fcd9e1cb'
==> Starting build()...
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type specified; using RelWithDebInfo
-- Using supported compiler GNU-7.3.0
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Checking for module 'cairo-fc'
--   Found cairo-fc, version 1.15.10
-- Checking for module 'alsa'
--   Found alsa, version 1.1.5
-- Checking for module 'libcurl'
--   Found libcurl, version 7.58.0
-- Found Libiw: /usr/lib/libiw.so  
-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.13
-- Checking for module 'xcb-randr'
--   Found xcb-randr, version 1.13
-- Checking for module 'xcb-randr>=1.12'
--   Found xcb-randr, version 1.13
-- Checking for module 'xcb-xrm'
--   Found xcb-xrm, version 1.2
-- Checking for module 'xcb-cursor'
--   Found xcb-cursor, version 0.1.3
--  Build:
--    Type: RelWithDebInfo
--    CC: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -O2 -g -DNDEBUG
--    CXX: /usr/bin/c++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wall -Wextra -Werror -Wno-noexcept-type -O2 -pedantic -pedantic-errors -O2 -g -DNDEBUG
--    LD: /usr/bin/ld -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 
--  Targets:
-- [X]   polybar-msg
-- [ ]   testsuite
--  Module support:
-- [X]   alsa
-- [X]   curl
-- [X]   i3
-- [ ]   mpd
-- [X]   network
--  X extensions:
-- [X]   xcb-randr
-- [X]   xcb-randr (monitor support)
-- [ ]   xcb-render
-- [ ]   xcb-damage
-- [ ]   xcb-sync
-- [ ]   xcb-composite
-- [X]   xcb-xkb
-- [X]   xcb-xrm
-- [X]   xcb-cursor
-- Font not found: fixed:pixelsize=10
-- Font not found: unifont:fontformat=truetype
-- Font not found: siji:pixelsize=10
-- Checking for module 'xcb-proto'
--   Found xcb-proto, version 1.13
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.14", minimum required is "2.7") 
-- XCB[XCB]: Found component XCB
-- Found XCB_XCB: /usr/lib/libxcb.so  
-- XCB[ICCCM]: Found component ICCCM
-- Found XCB_ICCCM: /usr/lib/libxcb-icccm.so  
-- XCB[EWMH]: Found component EWMH
-- Found XCB_EWMH: /usr/lib/libxcb-ewmh.so  
-- XCB[UTIL]: Found component UTIL
-- Found XCB_UTIL: /usr/lib/libxcb-util.so  
-- XCB[IMAGE]: Found component IMAGE
-- Found XCB_IMAGE: /usr/lib/libxcb-image.so  
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so  
-- Searching for xcbgen with python2
-- Searching for xcbgen with python3
-- Found xcbgen in /usr/lib/python3.6/site-packages
-- XCB[RANDR]: Found component RANDR
-- Found XCB_RANDR: /usr/lib/libxcb-randr.so  
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so;/usr/lib/libxcb-randr.so  
-- XCB[XKB]: Found component XKB
-- Found XCB_XKB: /usr/lib/libxcb-xkb.so  
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so;/usr/lib/libxcb-randr.so;/usr/lib/libxcb-xkb.so  
-- xpp: including xcb proto randr.xml
-- xpp: including xcb proto xkb.xml
-- xpp: including xcb proto xproto.xml
-- Checking for module 'jsoncpp>=1.7.7'
--   Found jsoncpp, version 1.8.4
-- Configuring done
-- Generating done
-- Build files have been written to: /home/tyler/builds/polybar/src/polybar/build
Scanning dependencies of target xpp
[  1%] Linking CXX static library libxpp.a
Traceback (most recent call last):
  File "/home/tyler/builds/polybar/src/polybar/lib/xpp/generators/cpp_client.py", line 3163, in <module>
    from xcbgen.state import Module
  File "/usr/lib/python3.6/site-packages/xcbgen/state.py", line 7, in <module>
    from xcbgen import matcher
  File "/usr/lib/python3.6/site-packages/xcbgen/matcher.py", line 12, in <module>
    from xcbgen.xtypes import *
  File "/usr/lib/python3.6/site-packages/xcbgen/xtypes.py", line 1221, in <module>
    class EventStruct(Union):
  File "/usr/lib/python3.6/site-packages/xcbgen/xtypes.py", line 1239, in EventStruct
    out = __main__.output['eventstruct']
KeyError: 'eventstruct'
make[2]: *** [lib/xpp/CMakeFiles/xpp.dir/build.make:70: lib/xpp/libxpp.a] Error 1
make[1]: *** [CMakeFiles/Makefile2:403: lib/xpp/CMakeFiles/xpp.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...

I am getting the same issue. Here is my output. 

```
 seth@nirvana  ~/Documents/polybar   master  makepkg -si
==> Making package: polybar 3.1.0-2 (Sun Mar 11 16:00:18 EDT 2018)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Cloning polybar git repo...
Cloning into bare repository '/home/seth/Documents/polybar/polybar'...
remote: Counting objects: 13709, done.
remote: Compressing objects: 100% (147/147), done.
remote: Total 13709 (delta 151), reused 101 (delta 74), pack-reused 13488
Receiving objects: 100% (13709/13709), 7.19 MiB | 11.20 MiB/s, done.
Resolving deltas: 100% (10261/10261), done.
==> Validating source files with md5sums...
    polybar ... Skipped
==> Extracting sources...
  -> Creating working copy of polybar git repo...
Cloning into 'polybar'...
done.
Switched to a new branch 'makepkg'
==> Starting prepare()...
Submodule 'lib/i3ipcpp' (https://github.com/jaagr/i3ipcpp) registered for path 'lib/i3ipcpp'
Submodule 'lib/xpp' (https://github.com/jaagr/xpp) registered for path 'lib/xpp'
Cloning into '/home/seth/Documents/polybar/src/polybar/lib/i3ipcpp'...
Cloning into '/home/seth/Documents/polybar/src/polybar/lib/xpp'...
Submodule path 'lib/i3ipcpp': checked out 'a6aa7a19786bdf7b96a02900510b3b3c325c8bdf'
Submodule path 'lib/xpp': checked out 'c73db50f3020b7abe294631b25a2d302fcd9e1cb'
==> Starting build()...
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type specified; using RelWithDebInfo
-- Using supported compiler GNU-7.3.0
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Checking for module 'cairo-fc'
--   Found cairo-fc, version 1.15.10
-- Checking for module 'alsa'
--   Found alsa, version 1.1.5
-- Checking for module 'libcurl'
--   Found libcurl, version 7.58.0
-- Found Libiw: /usr/lib/libiw.so
-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.13
-- Checking for module 'xcb-randr'
--   Found xcb-randr, version 1.13                                                                                                      
-- Checking for module 'xcb-randr>=1.12'                                                                                                
--   Found xcb-randr, version 1.13
-- Checking for module 'xcb-xrm'
--   Found xcb-xrm, version 1.2
-- Checking for module 'xcb-cursor'
--   Found xcb-cursor, version 0.1.3
--  Build:
--    Type: RelWithDebInfo
--    CC: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -O2 -g -DNDEBUG
--    CXX: /usr/bin/c++ -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wall -Wextra -Werror -Wno-noexcept-typ
e -O2 -pedantic -pedantic-errors -O2 -g -DNDEBUG
--    LD: /usr/bin/ld -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 
--  Targets:
-- [X]   polybar-msg
-- [ ]   testsuite
--  Module support:
-- [X]   alsa
-- [X]   curl
-- [X]   i3
-- [ ]   mpd
-- [X]   network
--  X extensions:
-- [X]   xcb-randr
-- [X]   xcb-randr (monitor support)
-- [ ]   xcb-render
-- [ ]   xcb-damage
-- [ ]   xcb-sync
-- [ ]   xcb-composite
-- [X]   xcb-xkb
-- [X]   xcb-xrm
-- [X]   xcb-cursor
-- Font not found: fixed:pixelsize=10
-- Font not found: unifont:fontformat=truetype
-- Font not found: siji:pixelsize=10
-- Checking for module 'xcb-proto'
--   Found xcb-proto, version 1.13
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.14", minimum required is "2.7")
-- XCB[XCB]: Found component XCB
-- Found XCB_XCB: /usr/lib/libxcb.so
-- XCB[ICCCM]: Found component ICCCM
-- Found XCB_ICCCM: /usr/lib/libxcb-icccm.so
-- XCB[EWMH]: Found component EWMH
-- Found XCB_EWMH: /usr/lib/libxcb-ewmh.so
-- XCB[UTIL]: Found component UTIL
-- Found XCB_UTIL: /usr/lib/libxcb-util.so
-- XCB[IMAGE]: Found component IMAGE
-- Found XCB_IMAGE: /usr/lib/libxcb-image.so
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so
-- Searching for xcbgen with python2
-- Searching for xcbgen with python3
-- Found xcbgen in /usr/lib/python3.6/site-packages
-- XCB[RANDR]: Found component RANDR
-- Found XCB_RANDR: /usr/lib/libxcb-randr.so
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so;/usr/
lib/libxcb-randr.so
-- XCB[XKB]: Found component XKB
-- Found XCB_XKB: /usr/lib/libxcb-xkb.so
-- Found XCB: /usr/lib/libxcb.so;/usr/lib/libxcb-icccm.so;/usr/lib/libxcb-ewmh.so;/usr/lib/libxcb-util.so;/usr/lib/libxcb-image.so;/usr/
lib/libxcb-randr.so;/usr/lib/libxcb-xkb.so
-- xpp: including xcb proto randr.xml
-- xpp: including xcb proto xkb.xml
-- xpp: including xcb proto xproto.xml
-- Checking for module 'jsoncpp>=1.7.7'
--   Found jsoncpp, version 1.8.4
-- Configuring done
-- Generating done
-- Build files have been written to: /home/seth/Documents/polybar/src/polybar/build
Scanning dependencies of target xpp
[  1%] Linking CXX static library libxpp.a
Traceback (most recent call last):
  File "/home/seth/Documents/polybar/src/polybar/lib/xpp/generators/cpp_client.py", line 3163, in <module>
    from xcbgen.state import Module
  File "/usr/lib/python3.6/site-packages/xcbgen/state.py", line 7, in <module>
    from xcbgen import matcher
  File "/usr/lib/python3.6/site-packages/xcbgen/matcher.py", line 12, in <module>
    from xcbgen.xtypes import *
  File "/usr/lib/python3.6/site-packages/xcbgen/xtypes.py", line 1221, in <module>
    class EventStruct(Union):
  File "/usr/lib/python3.6/site-packages/xcbgen/xtypes.py", line 1239, in EventStruct
    out = __main__.output['eventstruct']
KeyError: 'eventstruct'
make[2]: *** [lib/xpp/CMakeFiles/xpp.dir/build.make:70: lib/xpp/libxpp.a] Error 1
make[1]: *** [CMakeFiles/Makefile2:403: lib/xpp/CMakeFiles/xpp.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```
Running new install of arch, running on i3.

--- a/contrib/polybar.aur/PKGBUILD
+++ b/contrib/polybar.aur/PKGBUILD
@@ -2,7 +2,7 @@
 # Contributor: Michael Carlberg <c@rlberg.se>
 pkgname=polybar
 pkgver=3.1.0
-pkgrel=2
+pkgrel=3
 pkgdesc="A fast and easy-to-use status bar"
 arch=("i686" "x86_64")
 url="https://github.com/jaagr/polybar"
@@ -23,6 +23,7 @@ md5sums=("SKIP")
 
 prepare() {
   git -C "${pkgname}" submodule update --init --recursive
+  git -C "${pkgname}/lib/xpp" checkout 00165e1a6d5dd61bc153e1352b21ec07fc81245d
   mkdir -p "${pkgname}/build"
 }
 

